dms/models/directory.py

```diff
@@ -643,17 +630,17 @@ def write(self, vals):
             res = super().write(vals)
 
         if self and any(
-            field for field in vals if field in ["root_storage_id", "parent_id"]
+            field for field in vals if field in ["storage_id", "parent_id"]
         ):
             records = self.sudo().search([("id", "child_of", self.ids)]) - self
-            if "root_storage_id" in vals:
-                records.write({"storage_id": vals["root_storage_id"]})
-            elif "parent_id" in vals:
+            if "parent_id" in vals:
                 parent = self.browse([vals["parent_id"]])
                 data = next(iter(parent.sudo().read(["storage_id"])), {})
                 records.write(
                     {"storage_id": self._convert_to_write(data).get("storage_id")}
                 )
+            elif "parent_id" not in vals and "storage_id" in vals:
+                records.write({"storage_id": vals["storage_id"]})
```

Comment:
This is writing again the same thanon the `super`

Reply:
It's necessary to prevent error in `test_storage` test (apply `storage_id` to all child directories).

Reply:
OK, I see, but this will trigger N * X writes if there are N levels of depth, and each level has X children. You should prevent this with a context.